tests/lib/nested.sh

```diff
@@ -68,6 +68,49 @@ nested_wait_for_snap_command() {
     done
 }
 
+nested_check_unit_active() {
+    local nested_unit="${1:-$NESTED_VM}"
+    local retry=${2:-5}
+    local wait=${3:-1}
+
+    while [ "$retry" -ge 0 ]; do
+        retry=$(( retry - 1 ))
+
+        if ! systemctl is-active "$nested_unit"; then
+            echo "Unit $nested_unit is not active. Aborting!"
+            return 1
+        fi
+        sleep "$wait"
```

Comment:
Is this retry loop meant to wait until the until the unit becomes active? It looks like it's going to give up on first iteration without actually retrying unless I'm missing something?